### ISSUE TITLE
rack-test finding class on body tag failing under CI but not locally

### ISSUE THREAD

**Issue by gtd** [jnicklas/capybara#1758]
I have a strange failure with the default driver, but only on CircleCI, not on my vagrant dev box running Ubuntu.  The failure is that an assertion of a class being on the body tag does not work:

```
assert has_selector?('body') # passes
assert has_selector?('body.splash') # fails
```

`puts body.html` clearly indicates the class is present.  Also changing the driver to `:webkit` makes it pass.  

I realize this is probably not enough to analyze the problem, but I'm not sure what the next step is to debug this problem.  Or maybe it rings a lucky bell and you could point me in the right direction?

This is using capybara 2.7.1, capybara-webkit 1.11.1, rails 4.0.13, nokogiri 1.5.11.

**Comment by twalpole** [comment]
Nokogiri 1.5.11 is 3 years old and had tons of issues in parsing and processing xml.  Since that's what Capybara uses to find elements when using the rack-test driver I would first try updating to a newer version of nokogiri

**Comment by gtd** [comment]
Yeah I had to do a bit of work to update other dependencies to get nokogiri updated, but it is indeed fixed in 1.6.8.  Thanks for the quick response.
